test/testdata/desugar/block_given.rb.desugar-tree.exp

```diff
@@ -20,22 +20,10 @@ class <emptyTree><<C <root>>> < (::<todo sym>)
       else
         false
       end
-      if the_block_name
-        ::<root>::<C Kernel>.block_given?()
-      else
-        false
-      end
+      ::<root>::<C Kernel>.block_given?()
```

Comment:
This change exposes a latent bug, that `isCallToBlockGivenP()` only detects `Kernel`, but not `::Kernel`.

Fixed up-stack in #9722